### ISSUE TITLE
Use RequireJS

### ISSUE THREAD

**Issue by lightsofapollo** [kewisch/ical.js#23]

**Comment by lightsofapollo** [comment]
@kewisch I have started on this, should have something end of the week.

**Comment by lightsofapollo** [comment]
I worked on this a bit and quickly found we have many circular reference issues when using requirejs. I think the benefits of using requirejs are really clear when trying to use ICAL.js in node particularly but I am not sure of how it helps xpcom usage ? I want to do this but not until we get some better test coverage.

As it stands I am not sure it makes sense to publish ICAL.js on npm because it populates a global variable.
That does not matter much for my (or I think lightnings needs) but its a nice to have eventually.

**Comment by kewisch** [comment]
requireJS doesn't help XPCOM at all, there is no standard loader I know of. For Lightning what I am doing now is packing the ICAL.js script into a .jsm file and exporting the ICAL object.

**Comment by lightsofapollo** [comment]
Going to close this out... I don't really care about converting unless it benefits us in a major way.
